pkg/api/types.go

```diff
@@ -3109,6 +3109,8 @@ const (
 	NodeMemoryPressure NodeConditionType = "MemoryPressure"
 	// NodeDiskPressure means the kubelet is under pressure due to insufficient available disk.
 	NodeDiskPressure NodeConditionType = "DiskPressure"
+	// NodeCPUPressure means the kubelet is under pressure due to insufficient available cpu.
```

Comment:
IIUC, this new condition won't be necessary if kubelet reserves one whole hardware thread by default right?
I suspect thats the default experience most users would want. Reserving that single core would let BE pods run without getting evicted, possibly squeezed on a single core by CFS though. 
CPU pressure as a condition is meant to indicate lack of available CPUs from a usage standpoint. A node under CPU pressure may not be ideal for Bu or BE pods since they will not make progress, or such pods may better be rescheduled.
I'd like to avoid introducing this new condition.

Reply:
@vishh -- Updated to remove CPU pressure node condition and evictions, require cpu reservation and manage the shared pool accordingly in the static policy.